portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/DataSourcesList.testcase

```diff
@@ -25,9 +25,25 @@ definition {
 	@ignore = "true"
 	@priority = "5"
 	test DataSourceListShowsAllDataSources {
+		for (var n : list "1,2,3") {
+			ACSettings.takeTokenFromAnalyticsCloud();
 
-		// TODO LRAC-8129 DataSourceListShowsAllDataSources
+			ACUtils.launchDXP();
 
+			var propertyName = ACDXPSettings.connectDXPtoAnalyticsCloud();
+
+			var dataSource = StringUtil.replace("${propertyName}", " Combined Property", "");
+
+			ACDXPSettings.disconnectDXPFromAnalyticsCloud();
+
+			ACUtils.launchAC();
+
+			ACProperties.switchProperty(propertyName = "${propertyName}");
+
+			ACNavigation.goToSettings();
+
+			ACSettings.assertDataSourceName(dataSourceName = "${dataSource}");
```

Comment:
@ViniLopes20 this checks that the data source name appears somewhere in the list after each loop.  I wonder, should there be another step to check that all three data sources are still in the list at the end? (Example: assert data source 1, data source 2, and data source 3 all exist)

What do you think @alexcandidos?

Reply:
Hey @christianstokes , makes sense! 

I was discussing with @ViniLopes20  to try to use ``StringUtil.add`` to store the dataSource names during the loop execution in a list, so it can be validated out of the loop in the end of the execution! 

:)

Reply:
I fixed it, thanks guys :)